data-prepper-plugins/s3-source/src/main/java/com/amazon/dataprepper/plugins/source/SqsWorker.java

```diff
@@ -168,6 +169,24 @@ else if (isEventNameCreated(entry.getValue().get(0))) {
         return deleteMessageBatchRequestEntryCollection;
     }
 
+    private Optional<DeleteMessageBatchRequestEntry> processS3Object(
+            final Map.Entry<Message, List<S3EventNotification.S3EventNotificationRecord>> entry,
+            final S3ObjectReference s3ObjectReference) {
+        Optional<DeleteMessageBatchRequestEntry> deleteMessageBatchRequestEntry = Optional.empty();
+        // SQS messages won't be deleted if we are unable to process S3Objects because of S3Exception: Access Denied
```

Comment:
It might be worth creating a javadoc comment for this method to raise the visibility of this comment.

Reply:
I'll make this change in the upcoming PR